tests/components/mqtt_statestream/test_init.py

```diff
@@ -3,375 +3,354 @@
-        mock_pub.reset_mock()
-
-        # Set a state of an entity
-        mock_state_change_event(self.hass, State("fake.entity", "on"))
-        self.hass.block_till_done()
-
-        # Make sure 'on' was published to pub/fake/entity/state
-        mock_pub.assert_called_with(self.hass, "pub/fake/entity/state", "on", 1, True)
-        assert mock_pub.called
-
-        mock_pub.reset_mock()
-        # Set a state of an entity that shouldn't be included
-        mock_state_change_event(self.hass, State("fake.entity2", "on"))
-        self.hass.block_till_done()
-
-        assert not mock_pub.called
-
-    @patch("homeassistant.components.mqtt.async_publish")
-    @patch("homeassistant.core.dt_util.utcnow")
-    def test_state_changed_event_include_domain_exclude_entity(
-        self, mock_utcnow, mock_pub
-    ):
-        """Test filtering with included domain and excluded entity."""
-        base_topic = "pub"
-
-        incl = {"domains": ["fake"]}
-        excl = {"entities": ["fake.entity2"]}
-
-        # Add the statestream component for publishing state updates
-        # Set the filter to allow fake.* items
-        assert self.add_statestream(
-            base_topic=base_topic, publish_include=incl, publish_exclude=excl
-        )
-        self.hass.block_till_done()
-
-        # Reset the mock because it will have already gotten calls for the
-        # mqtt_statestream state change on initialization, etc.
-        mock_pub.reset_mock()
-
-        # Set a state of an entity
-        mock_state_change_event(self.hass, State("fake.entity", "on"))
-        self.hass.block_till_done()
-
-        # Make sure 'on' was published to pub/fake/entity/state
-        mock_pub.assert_called_with(self.hass, "pub/fake/entity/state", "on", 1, True)
-        assert mock_pub.called
-
-        mock_pub.reset_mock()
-        # Set a state of an entity that shouldn't be included
-        mock_state_change_event(self.hass, State("fake.entity2", "on"))
-        self.hass.block_till_done()
-
-        assert not mock_pub.called
+async def add_statestream(
+    hass,
+    base_topic=None,
+    publish_attributes=None,
+    publish_timestamps=None,
+    publish_include=None,
+    publish_exclude=None,
+):
+    """Add a mqtt_statestream component."""
+    config = {}
+    if base_topic:
+        config["base_topic"] = base_topic
+    if publish_attributes:
+        config["publish_attributes"] = publish_attributes
+    if publish_timestamps:
+        config["publish_timestamps"] = publish_timestamps
+    if publish_include:
+        config["include"] = publish_include
+    if publish_exclude:
+        config["exclude"] = publish_exclude
+    return await async_setup_component(
+        hass, statestream.DOMAIN, {statestream.DOMAIN: config}
+    )
+
+
+async def test_fails_with_no_base(hass, mqtt_mock):
+    """Setup should fail if no base_topic is set."""
+    assert await add_statestream(hass) is False
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5745133)